scrapy/extensions/feedexport.py

```diff
@@ -186,86 +187,113 @@ def _store_in_thread(self, file):
         ftp.quit()
 
 
-class SpiderSlot(object):
-    def __init__(self, file, exporter, storage, uri):
+class FeedSlot(object):
+    def __init__(self, file, exporter, storage, uri, feed_format, store_empty):
         self.file = file
         self.exporter = exporter
         self.storage = storage
+        # feed params
         self.uri = uri
+        self.feed_format = feed_format
+        self.store_empty = store_empty
+        # flags
         self.itemcount = 0
+        self._exporting = False
 
 
 class FeedExporter(object):
 
-    def __init__(self, settings):
-        self.settings = settings
-        self.urifmt = settings['FEED_URI']
-        if not self.urifmt:
-            raise NotConfigured
-        self.format = settings['FEED_FORMAT'].lower()
-        self.export_encoding = settings['FEED_EXPORT_ENCODING']
-        self.storages = self._load_components('FEED_STORAGES')
-        self.exporters = self._load_components('FEED_EXPORTERS')
-        if not self._storage_supported(self.urifmt):
-            raise NotConfigured
-        if not self._exporter_supported(self.format):
-            raise NotConfigured
-        self.store_empty = settings.getbool('FEED_STORE_EMPTY')
-        self._exporting = False
-        self.export_fields = settings.getlist('FEED_EXPORT_FIELDS') or None
-        self.indent = None
-        if settings.get('FEED_EXPORT_INDENT') is not None:
-            self.indent = settings.getint('FEED_EXPORT_INDENT')
-        uripar = settings['FEED_URI_PARAMS']
-        self._uripar = load_object(uripar) if uripar else lambda x, y: None
-
     @classmethod
     def from_crawler(cls, crawler):
-        o = cls(crawler.settings)
-        o.crawler = crawler
+        o = cls(crawler)
         crawler.signals.connect(o.open_spider, signals.spider_opened)
         crawler.signals.connect(o.close_spider, signals.spider_closed)
         crawler.signals.connect(o.item_scraped, signals.item_scraped)
         return o
 
+    def __init__(self, crawler):
+        self.crawler = crawler
+        self.settings = crawler.settings
+        self.feeds = {}
+        self.slots = []
+
+        if not self.settings['FEEDS'] and not self.settings['FEED_URI']:
+            raise NotConfigured
+
+        FEEDS = self.settings['FEEDS'] or {}
+        for uri, feed in FEEDS.items():
+            self.feeds[uri] = feed_complete_default_values_from_settings(
+                {key.strip().upper(): value for key, value in feed.items()},
+                self.settings
+            )
+
+        # Begin: Backward compatibility for FEED_URI and FEED_FORMAT settings
+        if not self.feeds and self.settings['FEED_URI']:
```

Comment:
I think a warning about `self.settings['FEED_URI']` should be issued regardless of whether or not `self.settings['FEEDS']` is also used.